File: BIZ/system/logic/code/dsrc/KSQLProcedure.java


import java.util.Map;

import com.justep.message.CommonMessages;
import com.justep.model.ModelUtils;
import com.justep.system.data.KSQL;
import com.justep.system.data.Table;
import com.justep.util.Utils;

public class KSQLProcedure {

	public static Table query(String ksql, Map<String, Object> variables, String dataModel, String fnModel) {
	
		Utils.check(Utils.isNotEmptyString(ksql) && Utils.isNotEmptyString(dataModel),
				CommonMessages.class, CommonMessages.PARAM_NULL1, "ksql, dataModel");
		return KSQL.select(ksql, variables, dataModel, Utils.isEmptyString(fnModel) ? null : ModelUtils.getModel(fnModel));

	}
}

File: BIZ/system/logic/action/ksql.procedure.m
<model xmlns="http://www.justep.com/model"
	xmlns:m="http://www.justep.com/model">
	<procedure name="ksqlQueryProcedure"
		code-model="/system/logic/code"
		code="KSQLProcedure.query">
		<parameter name="ksql" type="String"/>
		<parameter name="variables" type="Map"/>
		<parameter name="dataModel" type="String"/>
		<parameter name="fnModel" type="String"/>
	</procedure>
</model>